traitsui/tests/test_ui_panel.py

```diff
@@ -55,3 +75,25 @@ def test_grouped_layout_with_springy(self):
         # This should not fail.
         with create_ui(obj1, dict(view=view)):
             pass
+
+    # Regression test for enthought/traitsui#1538
+    @unittest.skipIf(no_modal_dialog_tester, "ModalDialogTester unavailable")
+    def test_show_help(self):
+
+        # This help window is not actually modal, when opened it will be the
+        # active window not active modal widget
+        class MyTester(ModalDialogTester):
+            def get_dialog_widget(self):
+                return self._qt_app.activeWindow()
```

Comment:
FWIW the test passed for me on mac and ubuntu locally without this _but_ without this the call of `x.click_button(OK)` effectively ends up doing nothing as we are trying to call a method on None.

This way we are actually closing the window we open which feels like the right thing to do.

The real cause of the test hang I believe was not having `auto_process_events=False`.  I should have realized this much faster than I did.  Previously I had only remembered this being necessary when using `UITester` _inside_ the use of the ModalDialogTester callback.  
For example:
https://github.com/enthought/traitsui/blob/62033f5e1319d349bb3e980d9083f2c2a24a431a/traitsui/tests/editors/test_instance_editor.py#L211-L243

In cases where we just use UI Tester to trigger the dialog being opened, it has not been needed (until now at least AFAICR) 
e.g.
https://github.com/enthought/traitsui/blob/62033f5e1319d349bb3e980d9083f2c2a24a431a/traitsui/examples/demo/Standard_Editors/tests/test_ButtonEditor_demo.py#L42-L69

Nonetheless, this test does what we want it to now and test suite seems to not be hanging

Reply:
But now this test is Qt-specific, without being marked as such

Reply:
I think it implicitly will be skipped if there is no qt as modal dialog tester only exists for qt currently, but you are right we should definitely be explicit here in saying this is a very qt specific test.
Will update this now